BEIMA.Backend.Test/DeviceFunctions/UpdateDeviceTest.cs

```diff
@@ -97,6 +125,7 @@ public async Task ExistingDevice_UpdatesDevice_ReturnsOKResponse()
             Assert.IsNotNull(response);
             Assert.That(response, Is.TypeOf(typeof(OkObjectResult)));
             Assert.That(((OkObjectResult)response).StatusCode, Is.EqualTo((int)HttpStatusCode.OK));
+            Assert.DoesNotThrow(() => mockStorage.Verify(mock => mock.DeleteFile(It.IsAny<string>()), Times.Exactly(2)));
```

Comment:
We should have a mock method for DeleteFile if we are going to be asserting on it. Also as a quick note, on the mocks you can have a double returns which will allow you to let it return two values for different inputs, or if you are expecting two different inputs, you can make two separate mock methods to run depending on the input. Sorry if this is a bit confusing.